packages/astro/src/runtime.ts

```diff
@@ -81,8 +81,8 @@ function getParams(array: string[]) {
 }
 
 async function getStaticPathsMemoized(runtimeConfig: AstroRuntimeConfig, component: string, mod: any, args: GetStaticPathsArgs): Promise<GetStaticPathsResult> {
-  runtimeConfig.cache.staticPaths[component] = runtimeConfig.cache.staticPaths[component] || (await mod.exports.getStaticPaths(args)).flat();
-  return runtimeConfig.cache.staticPaths[component];
+  runtimeConfig.cache.staticPaths[component] = runtimeConfig.cache.staticPaths[component] || mod.exports.getStaticPaths(args);
```

Comment:
nit: can you update the type for `runtimeConfig.cache.staticPaths`? Now that this is storing promises instead of objects.

Reply:
that's a good point - my ide was resolving the type as any 😅 so i'll try to sort that out

Reply:
if it's a larger issue of us having `cache` be untyped, not required to solve it here

Reply:
Updated 

https://github.com/snowpackjs/astro/blob/1bc357d38091a02189e9bf34898c69bbfc82c773/packages/astro/src/runtime.ts#L34-L34

but `mod.exports.getStaticPaths()` is typed as `any`, so it's unclear exactly how useful this is here 😅 this is where I run into distrust of Typescript's typesystem when it doesn't provide runtime type checks 😔

Reply:
agreed, but Promise vs. not promise is a super useful distinction, and can help catch issues where you forget to await it later on